base/heartbeat.go

```diff
@@ -83,12 +84,13 @@ type couchbaseHeartBeater struct {
 // the keyPrefix which will be prepended to the heartbeat doc keys,
 // and the nodeUUID, which is an opaque identifier for the "thing" that is using this
 // library.  nodeUUID will be passed to listeners on stale node detection.
-func NewCouchbaseHeartbeater(bucket Bucket, keyPrefix, nodeUUID string) (heartbeater *couchbaseHeartBeater, err error) {
+func NewCouchbaseHeartbeater(bucket Bucket, keyPrefix, nodeUUID string, groupID *string) (heartbeater *couchbaseHeartBeater, err error) {
```

Comment:
Looking at use cases like this one - where the functionality already accepts a key prefix - has me thinking about the relative merits of _sync:[groupID]:[functionalPrefix]:[value] versus _sync:[functionalPrefix]:[groupID]:[value], in particular for cases such as filtering a DCP stream.  The former would make it easier to broadly filter by config group, but a bit more difficult to filter by functional prefix.  Was there any discussion of the relative merits of each approach?

Reply:
There was a discussion on this, and the reason it was decided that it would be best to include the group ID after the functional prefix is because of the `DocChanged` function for the DCP filtering prefix checks.

However, due to the naming collision mention in the other comment, the group ID will be used before the functional prefix if a group ID is being used (or else the old legacy doc name will be used).